clang/lib/Sema/SemaOverload.cpp

```diff
@@ -16333,7 +16387,7 @@ ExprResult Sema::FixOverloadedFunctionReference(Expr *E, DeclAccessPair Found,
     assert(UnOp->getOpcode() == UO_AddrOf &&
            "Can only take the address of an overloaded function");
     if (CXXMethodDecl *Method = dyn_cast<CXXMethodDecl>(Fn)) {
-      if (Method->isStatic()) {
+      if (!Method->isImplicitObjectMemberFunction()) {
         // Do nothing: static member functions aren't any different
         // from non-member functions.
```

Comment:
nit: maybe update the comment too to indicate that this is also intended to handle explicit object member functions.